### PR TITLE
Http misc

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: contrib/epee/include/net/http_protocol_handler.inl [hunk 1]
```diff
@@ -351,13 +358,18 @@ namespace net_utils
...
 	{ 
-		STATIC_REGEXP_EXPR_1(rexp_match_command_line, "^(((OPTIONS)|(GET)|(HEAD)|(POST)|(PUT)|(DELETE)|(TRACE)) (\\S+) HTTP/(\\d+).(\\d+))\r?\n", boost::regex::icase | boost::regex::normal);
+		STATIC_REGEXP_EXPR_1(rexp_match_command_line, "^(((OPTIONS)|(GET)|(HEAD)|(POST)|(PUT)|(DELETE)|(TRACE)) (\\S+) HTTP/(\\d+)\\.(\\d+))\r?\n", boost::regex::icase | boost::regex::normal);
 		//											    123         4     5      6      7     8        9        10          11     12    
 		//size_t match_len = 0;
 		boost::smatch result;	
 		if(boost::regex_search(m_cache, result, rexp_match_command_line, boost::match_default) && result[0].matched)
 		{
-			analize_http_method(result, m_query_info.m_http_method, m_query_info.m_http_ver_hi, m_query_info.m_http_ver_hi);
+			if (!analize_http_method(result, m_query_info.m_http_method, m_query_info.m_http_ver_hi, m_query_info.m_http_ver_hi))
```
Comment (anonimal) [inline contrib/epee/include/net/http_protocol_handler.inl]: `analyze_http_method`. Otherwise it sounds like something done to an anal.

Comment (vtnerd) [reply contrib/epee/include/net/http_protocol_handler.inl]: This is one of my favorite cryptonote misspellings.

File: contrib/epee/include/net/http_protocol_handler.inl [hunk 2]
```diff
@@ -328,8 +328,15 @@ namespace net_utils
 	inline bool analize_http_method(const boost::smatch& result, http::http_method& method, int& http_ver_major, int& http_ver_minor)
 	{
 		CHECK_AND_ASSERT_MES(result[0].matched, false, "simple_http_connection_handler::analize_http_method() assert failed...");
-		http_ver_major = boost::lexical_cast<int>(result[11]);
-		http_ver_minor = boost::lexical_cast<int>(result[12]);
+		try
```
Comment (vtnerd) [inline contrib/epee/include/net/http_protocol_handler.inl]:
```c++
if (!boost::conversion::try_lexical_convert(result[11], http_ver_major))
    return false;
```
Is available in Boost 1.57+. Internally, `boost::lexical_cast` is doing this call anyway.

Comment (moneromooo-monero) [reply contrib/epee/include/net/http_protocol_handler.inl]: Thank, looks like the min boost version for monero is now 1.58, I'll change it.

**Pull request merged**